zeppelin-web/src/app/credential/credential.controller.js

```diff
@@ -33,24 +48,118 @@ angular.module('zeppelinWebApp').controller('CredentialCtrl', function($scope, $
     }
 
     $http.put(baseUrlSrv.getRestApiBase() + '/credential',
-      { 'entity': $scope.credentialEntity,
-        'username': $scope.credentialUsername,
-        'password': $scope.credentialPassword
-      } ).
+    {
+      'entity': $scope.entity,
+      'username': $scope.username,
+      'password': $scope.password
+    }).
     success(function (data, status, headers, config) {
       BootstrapDialog.alert({
         closable: true,
         message: 'Successfully saved credentials.'
       });
-      $scope.credentialEntity = '';
-      $scope.credentialUsername = '';
-      $scope.credentialPassword = '';
+      getCredentialInfo();
+      $scope.resetCredentialInfo();
+      $scope.showAddNewCredentialInfo = false;
       console.log('Success %o %o', status, data.message);
     }).
     error(function (data, status, headers, config) {
-      alert('Error saving credentials');
+      BootstrapDialog.alert({
+        closable: true,
+        message: 'Error saving credentials'
+      });
       console.log('Error %o %o', status, data.message);
     });
   };
 
+  $scope.cancelCredentialInfo = function() {
+    $scope.showAddNewCredentialInfo = false;
+    $scope.resetCredentialInfo();
+  };
+
+  $scope.resetCredentialInfo = function() {
+    $scope.entity = '';
+    $scope.username = '';
+    $scope.password = '';
+  };
+
+  $scope.copyOriginCredentialsInfo = function(entity) {
+    var index = _.findIndex($scope.credentialInfo, { 'entity': entity });
+    credentialInfoTmp[index] = angular.copy($scope.credentialInfo[index]);
+
+    ngToast.info({
+      content: 'Since entity is a unique key, you can edit only username & password',
+      verticalPosition: 'bottom',
+      timeout: '3000'
+    });
+  };
+
+  $scope.updateCredentialInfo = function(form, entity, username, password) {
+    BootstrapDialog.confirm({
```

Comment:
A bootstrap confirmation isn't needed here, the edit mode already had save or cancel buttons